backoffice/modules/catalog/components/CategoryMapping.tsx

```diff
@@ -3,20 +3,116 @@ import { UseFormSetValue, UseFormGetValues } from 'react-hook-form';
 import { useEffect, useState } from 'react';
 import { getCategories } from '../services/CategoryService';
 import { Category } from '../models/Category';
-
+import { useRouter } from 'next/router';
+import { getProduct } from '../services/ProductService';
+import { Product } from '../models/Product';
 type Props = {
   setValue: UseFormSetValue<ProductPost>;
   getValue: UseFormGetValues<ProductPost>;
 };
 const ProductCategoryMapping = ({ setValue, getValue }: Props) => {
   const [categories, setCategories] = useState<Category[]>([]);
+  let [checkCategory, setCheckCategory] = useState<string[]>([]);
+  let listCheckCategory: string[] = [];
+  let childrenOfCategoryId: string[] = [];
+  let parentOfCategoryId: string[] = [];
+  let categoryIds: number[] = [];
+  const [product, setProduct] = useState<Product>();
+
+  //Get ID
+  const router = useRouter();
```

Comment:
Can we get the product in the page and pass it down to components?

Reply:
no, since CreateProduct and UpdateProduct both use it, when getting the product in the page and passing it to the components, the Create part will be hard to handle

Reply:
The create product page can create an empty product and pass it down just like in the case of edit?